modules/apps/roles/roles-selector-web/src/main/java/com/liferay/roles/selector/web/internal/display/context/EditRolesUsersDisplayContext.java

```diff
@@ -52,78 +54,85 @@ public SearchContainer<User> getSearchContainer() throws PortalException {
 			return _userSearch;
 		}
 
-		_userSearch = new UserSearch(
-			_renderRequest,
-			(PortletURL)_httpServletRequest.getAttribute(
-				"edit_roles.jsp-portletURL"));
-
-		UserSearchTerms searchTerms =
-			(UserSearchTerms)_userSearch.getSearchTerms();
-
-		LinkedHashMap<String, Object> userParams =
-			LinkedHashMapBuilder.<String, Object>put(
-				"inherit", Boolean.TRUE
-			).put(
-				"usersGroups", Long.valueOf(_getGroupId())
-			).put(
-				"userGroupRole",
-				() -> {
-					if (Objects.equals(
-							_httpServletRequest.getAttribute(
-								"edit_roles.jsp-tabs1"),
-							"current")) {
-
-						return new Long[] {
-							Long.valueOf(_getGroupId()),
-							Long.valueOf(_getRoleId())
-						};
+		if (GroupPermissionUtil.contains(
```

Comment:
buy space and now we are returning null in this scenario. Is it correct? (I mean... does it break something?)

Reply:
From my testing it only 'breaks' if you follow the reproduction steps, intercept the request and don't have permission. So for example:
1. Omniadmin user has access to both organizations and even with intercepted request, still is able to see either of them (due to them technically having permission to see both).
2. User who is an organization admin sees the list properly if they're accessing their organization, but sees a "you don't have permission" message if they attempt to access the list of a group they don't have user assign roles access to.

The logic is similar to what we see in OrganizationsManagementToolbarDisplayContext.getActionDropdownItems() where we also return null. I could also add the permission check in edit_roles_users.jsp instead if preferred.